parallel-install/pkg/config/kubeconfigmanager.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeConfigManager handles kubeconfig path and kubeconfig content.
+type kubeConfigManager struct {
+	path    string
+	content string
+}
+
+// NewKubeConfigManager creates a new instance of KubeConfigManager.
+func NewKubeConfigManager(path, content *string) (*kubeConfigManager, error) {
+	pathExists := exists(path)
+	contentExists := exists(content)
+	var resolvedPath string
+	var resolvedContent string
+
+	if pathExists && contentExists {
+		resolvedPath = *path
+	} else if pathExists {
+		resolvedPath = *path
+	} else if contentExists {
+		resolvedContent = *content
+	} else {
+		return nil, errors.New("either kubeconfig or kubeconfigcontent property has to be set")
+	}
+
+	return &kubeConfigManager{
+		path:    resolvedPath,
+		content: resolvedContent,
+	}, nil
+}
+
+// Path returns a path to the kubeconfig file.
+func (k *kubeConfigManager) Path() (string, error) {
+	return k.resolvePath()
+}
+
+// Config returns a kubeconfig REST Config used by k8s clients.
+func (k *kubeConfigManager) Config() (*rest.Config, error) {
+	if exists(&k.path) {
+		resolvedPath, err := k.resolvePath()
+		if err != nil {
+			return nil, err
+		}
+
+		return clientcmd.BuildConfigFromFlags("", resolvedPath)
+	} else {
+		return clientcmd.RESTConfigFromKubeConfig([]byte(k.content))
+	}
+}
+
+func (k *kubeConfigManager) resolvePath() (path string, err error) {
```

Comment:
this is a bit confusing for me cause i would say that creating temp file shouldn't be responsibility of this method. especially in case it's usage in Config() method

Reply:
Ok - a separate function for creating a file and the second one to create a final path seems decent.